testutil/testclient/testqueryclients/grpcserver.go

```diff
@@ -0,0 +1,168 @@
+package testqueryclients
+
+import (
+	"context"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+)
+
+// callCounter is a simple struct that keeps track of the number of times a method is called
+type callCounter struct {
+	callCount int
+}
+
+func (c *callCounter) CallCount() int {
+	return c.callCount
+}
+
+func (c *callCounter) Increment() {
+	c.callCount++
+}
+
+// MockServiceQueryServer is a mock implementation of the servicetypes.QueryServer interface
```

Comment:
Up until now we've always been using `gomock` to generate this sort of thing, which has support for call counters. 

I feel strongly that we should not be changing patterns now.